snorkel/map/core.py

```diff
@@ -28,19 +33,90 @@ def get_parameters(
     return params[0]
 
 
+def is_hashable(obj: Any) -> bool:
+    """Test if object is hashable via duck typing.
+
+    NB: not using `collections.Hashable` as some objects
+    (e.g. pandas.Series) have a `__hash__` method to throw
+    a more specific exception.
+    """
+    try:
+        hash(obj)
+        return True
+    except Exception:
+        return False
+
+
+def get_hashable(obj: Any) -> Hashable:
```

Comment:
FWIW: If we did the approach where we converted every data type to a common IR, we could just make sure that said IR was hashable with `hash` and skip this complexity, which makes me nervous...

Reply:
(this is not a minor comment, but not an actionable one in this PR...)

Reply:
Not necessarily...the values in certain fields of the IR might not be hashable (e.g. a dictionary). I agree this would help with the top-layer, but we'd still need this functionality. Will keep it on the roadmap.